src/main/org/codehaus/groovy/control/ResolveVisitor.java

```diff
@@ -852,6 +858,21 @@ protected Expression transformPropertyExpression(PropertyExpression pe) {
         return ret;
     }
 
+    private boolean isVisibleNestedClass(ClassNode type, ClassNode ceType) {
+        if (!type.isRedirectNode()) return false;
+        ClassNode redirect = type.redirect();
+        // public or private
+        if (Modifier.isPublic(redirect.getModifiers()) || Modifier.isProtected(redirect.getModifiers())) return true;
```

Comment:
The comment is misleading (private -> protected), but the code reads very good so maybe no comment needed.